lib/token.h

```diff
@@ -143,6 +145,8 @@ struct TokenImpl {
     void setCppcheckAttribute(CppcheckAttributes::Type type, MathLib::bigint value);
     bool getCppcheckAttribute(CppcheckAttributes::Type type, MathLib::bigint &value) const;
 
+    SmallVector<ReferenceToken>* mRefs{};
```

Comment:
If this is going to be a pointer, you should use `std::unique_ptr` or `std::shared_ptr`.

Reply:
I modeled it after `mValues` which is also just a raw pointer.

Reply:
You should model it after `mAttributeAlignas`, thats a much safer choice. I think the raw pointers were used before managed pointers were available.

Reply:
You can just move the `ReferenceToken` to the token.h header.

Reply:
True, but `followAllReferences()` is the code which produces the `ReferenceToken` objects so I would prefer to keep those together.

It is unfortunate enough that `smallvector.h` spilled into more code.

Reply:
The spillage could be addressed by moving `TokenImpl` into the implementation (hence the name). I tried that in the past but unfortunately the code relies on the implementation being exposed - I will give it another try.